src/api/package/docs.ts

```diff
@@ -35,3 +36,43 @@ export const fetchMarkdown = async (
   }
   return response.text();
 };
+
+/**
+ * Fetch JSON docs of a specific package from the backend.
+ */
+export const fetchJsonDocs = async (
+  name: string,
+  version: string,
+  language: string,
+  scope?: string,
+  submodule?: string
+): Promise<Schema> => {
+  const docsSuffix = `/docs-${
+    submodule ? `${submodule}-` : ""
+  }${language}.json`;
+
+  const jsonPath = `${getAssetsPath(name, version, scope)}${docsSuffix}`;
+  console.log("jsonPath", jsonPath);
+  const response = await fetch(jsonPath);
+  if (!response.ok) {
+    throw new Error(
+      `Failed fetching documentation for ${jsonPath}: ${response.statusText}`
+    );
+  }
+
+  // since CloudFront returns a 200 /index.html for missing documents,
+  // we assert the expected docs content type to detect these errors.
+  // TODO: switch to proper 404 responses in this case (requires backend changes)
+  const expectedContentType = "application/json";
+  const contentType = response.headers.get("Content-Type");
+
+  // we check 'includes' and not 'equal' because the content type contains
+  // charset as well (e.g text/markdown; charset=UTF-8)
+  if (!contentType || !contentType.includes(expectedContentType)) {
+    throw new Error(
+      `Invalid content type: ${contentType}. Expected ${expectedContentType}"`
+    );
+  }
```

Comment:
The comments here are copied from the similar `fetchMarkdownDocs` function above.

Note - this is currently raising errors because of the bug caught in https://github.com/cdklabs/construct-hub/pull/740 -- once our data on constructs.dev gets reprocessed this should be fixed, but I'll make sure to validate this before merging